### PR TITLE
Add default values for properties used in the property pages

### PR DESCRIPTION
**Pull request by RaulPerez1**
https://github.com/dotnet/roslyn-project-system/issues/437

### CODE REVIEW

**Review by srivatsn — APPROVED** [review]

**Comment by mavasani** [comment]
Does this mean for any new Boolean property we add to a new or existing property page, we will need a similar change in this file to define the default?

I am adding 2 new checkboxes in Package property page in https://github.com/dotnet/roslyn-project-system/pull/950 and see same indeterminate state as default.

Can't we ask CPS to add support to defining defaults in the XAML rule files itself? This will avoid cross cutting changes.

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.props [hunk 1]
```diff
@@ -68,6 +68,12 @@ Copyright (c) .NET Foundation. All rights reserved.
     <!-- Exclude GAC, registry, output directory from search paths. -->
     <AssemblySearchPaths>{CandidateAssemblyFiles};{HintPathFromItem};{TargetFrameworkDirectory};{RawFileName}</AssemblySearchPaths>
     <DesignTimeAssemblySearchPaths>$(AssemblySearchPaths)</DesignTimeAssemblySearchPaths>
+
+    <AllowUnsafeBlocks Condition="'$(AllowUnsafeBlocks)'==''">false</AllowUnsafeBlocks>
+    <TreatWarningsAsErrors Condition="'$(TreatWarningsAsErrors)'==''">false</TreatWarningsAsErrors>
+    <Prefer32Bit Condition="'$(Prefer32Bit)'==''">false</Prefer32Bit>
+    <SignAssembly Condition="'$(SignAssembly)'==''">false</SignAssembly>
+    <DelaySign Condition="'$(DelaySign)'!=''">false</DelaySign>
```
Comment (mavasani) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.props]: This seems wrong - why are we default DelaySign to false when it already has a non-empty setting?

Comment (RaulPerez1) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.props]: The property pages are expecting a Boolean value and empty strings are throwing a wrench on things.

For your other question - I'll forward you the email threads with the discussion but basically yes they should be defaulted as it is the right thing from the project perspective.

Comment (mavasani) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.props]: Yes, but why is DelaySign defaulting conditional reverse of the ones above?

Comment (RaulPerez1) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.props]: Good catch, that's getting wiped away (details on the thread) so I missed that. I'll send out another PR with an update.

**Pull request merged**